app/experimenter/docs/adrs/0001-create-new-nimbus-application.md

```diff
@@ -0,0 +1,119 @@
+# Create A New Web Application To Manage Nimbus Experiments
+
+* Status: proposed
+* Deciders: Kate Hudson, Jared Lockhart, TBD
+* Date: 2020-08-18
+
+## Context and Problem Statement
+
+Experimenter is a web application developed solely for the purpose of managing product experiments at Mozilla.  It was originally designed to manage the Shield experimentation workflow, and has expanded to include a comprehensive workflow for managing Normandy experiments.  Project Nimbus is being developed with very different core assumptions than the previous experimentation platforms, and it is unclear whether building on top of the existing Experimenter infrastructure or creating a new purpose specific application will best suit the project's needs.
```

Comment:
Reading that "Project Nimbus is being developed with very different core assumptions" makes me want to see what the diff in assumptions are in order to know how that would affect the decision here.

Not very familiar with Experimenter, Shield, Normandy, or Nimbus. Some links to context might help - docs, talks, whatever might be available.

Reply:
I think the most relevant thing to note here is that prior to project Nimbus, Experimenter played a supporting role in the *planning* and *reporting* phases of some types of experiments (mainly for Firefox Desktop), but the configuration, launch, and review of experiments happens across many different tools (Normandy, Github/Remote Settings, or third-party tools like Leanplum, among others), most of which require expert operators and coordinators to be involved. 

In Nimbus, the central assumption is that the Experiment Console should play a much more central role as the single source of truth for all experiments. It should provide a single web interface for the configuration/management/review/launch of experiments across platforms, rather than having to be translated manually by operators and coordinators into different systems for each experiment type. It should enforce a single API and experiment format across platforms (mobile, desktop, and eventually web).

This [Vision Document](https://docs.google.com/document/d/10XFq-nWS76UBUsytHnwoaie4Kz9-zO8JrtMZzcaJ5h0/edit#) goes into more detail about why, this [system diagram of experiments in Firefox](https://docs.google.com/document/d/1ypWh-aug9NFdArQfYCuLYpUpK_IKdgELVmx21CaWI6w/edit#heading=h.s770289vwy3r) describes the system as it exists prior to nimbus, at least the part relevant to Desktop.

Reply:
From the docs I've seen on Nimbus, the architecture has some fundamental differences (typically simplifications) that:
- Change what service/UX is responsible for a given function.
- Consolidate or remove portions to streamline interaction and/or process.

I assume this is the 'very different core assumption' being referenced here?

Reply:
Yeah I think the broadest way I can describe it are as:

The old (Normandy focused) workflow is about capturing an experiment _proposal_ in high level natural language, and then many parties (like data science, firefox engineering, release management, qa) converting that into a concrete experiment implementation, analyzing and capturing risks, and making sure those are all signed off on.  It's a very involved workflow that touches many parties, and Experimenter's role is to capture and orchestrate all of it.

The new (Nimbus) workflow is about exposing a configuration panel that in theory a single experiment owner can use to define and launch an entire experiment without needing any of the steps I listed above.  The differences that allow this are greatly restricting and simplifying the experiment configuration space, Experimenter generating the experiment configuration (recipe) in its entirety, and sending that directly to Firefox through Remote Settings.

So the overall effect is that the majority of the UI that was built for the old workflow doesn't apply, but the core mechanics and logic in the backend internals still do.  The way we've approach that thus far is to create a new set of isolated APIs and React frontend for the new workflow that sits next to the old one, but sharing the same db models and logic.  It's worked well so far.

Reply:
Shield is an obsolete name for the Firefox experimentation program; "Shield studies" were/are shipped using Normandy.

Experimenter helps manage some (but not all) experimentation for our mobile products, which do not use Normandy because the client is not available. This is an example of an experiment deployed on Fenix using Mako (which is part of Android Components and which consumes content from Kinto), which had some metadata tracked in Experimenter: https://experimenter.services.mozilla.com/experiments/search-widget-discoverability/

Reply:
> This is an example of an experiment deployed on Fenix using Mako

Adding for completeness: replacing Mako is an explicit goal of the nimbus mobile SDK

Reply:
Normandy experiments (pref-flip and add-on experiments)
Message system (xman) [CFR](https://experimenter.services.mozilla.com/experiments/high-velocity-message-testing-multi-account-containers-retention-study/) + about:welcome experiments -[ex](https://experimenter.services.mozilla.com/experiments/multi-stage-aboutwelcome-v3-80-release/) 
Mobile experiments on Mako ([ex](https://experimenter.services.mozilla.com/experiments/search-widget-discoverability/))
Rollouts (add-on & [pref-flip](https://experimenter.services.mozilla.com/experiments/activity-stream-pocket-new-tab-spoc-personalization-v2-rollout/)) 
Using pref flips to add experimentation to other systems - Pocket ([ex](https://experimenter.services.mozilla.com/experiments/activity-stream-pocket-new-tab-story-spocs-in-uk-experiment/)) + snippets ([ex](https://experimenter.services.mozilla.com/experiments/pref-flip-experiment-high-velocity-message-testing-f100-snippets/))
Generic experiments for visibility or for someone needing a tool to help coordinate but not having one that fits. ([ex](https://experimenter.services.mozilla.com/experiments/youtube-regrets-reporter/)).  We use generic intake as we address new request for intake assistance as an easy way to test out scenario to see if custom work is needed.